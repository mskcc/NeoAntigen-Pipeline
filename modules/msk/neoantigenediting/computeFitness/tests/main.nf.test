nextflow_process {

    name "Test Process NEOANTIGENEDITING_COMPUTEFITNESS"
    script "../main.nf"
    process "NEOANTIGENEDITING_COMPUTEFITNESS"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "neoantigenediting"
    tag "neoantigenediting/computeFitness"

    test("neoantigenediting_computeFitness - json") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true)                   
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['iedb_alignments'], checkIfExists: true)                  
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.annotated_output[0][1]).name).match() }

            )
        }

    }

    test("neoantigenediting_computeFitness - json - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('patient_data')                   
                    ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('alignment_file')                  
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }


}
