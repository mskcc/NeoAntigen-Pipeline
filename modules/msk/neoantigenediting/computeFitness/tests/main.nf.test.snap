{
    "neoantigenediting_computeFitness - json": {
        "content": [
            "3_OLTS_primary_tumor_annotated.json"
        ],
        "timestamp": "2024-03-20T15:58:08.710924"
    },
    "neoantigenediting_computeFitness - json - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "patient_data_annotated.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e535d03b1366a3bee2e96c63a942b180"
                ],
                "annotated_output": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "patient_data_annotated.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e535d03b1366a3bee2e96c63a942b180"
                ]
            }
        ],
        "timestamp": "2024-03-20T15:58:14.290682"
    }
}